WooCommerce/Classes/ViewRelated/MainTabBarController.swift

```diff
@@ -183,6 +183,17 @@ final class MainTabBarController: UITabBarController {
         }
     }
 
+    override func viewSafeAreaInsetsDidChange() {
+        super.viewSafeAreaInsetsDidChange()
+
+        if tabBar.isHidden {
+            let bottomInset = 8.0
```

Comment:
I've also attempted to set the `bottomInset` by calculating it via `-tabBar.frame.height + 8` but doesn't seem to play nice with the `POSFloatingControlView` and the `Checkout` button, as both elements seem to set themselves at different heights if we do so 🤔

Reply:
Great questions!
1. I don't think there's a good reason. Most likely comes from having narrow vision after trying multiple approaches. I've updated it here https://github.com/woocommerce/woocommerce-ios/pull/13785/commits/2c287f4178b29f699286605cc476b5e2e806a2fc so we don't need the override
2. We do, if we set it to .zero then we see the problem above. I'm not really sure why, but perhaps internal optimization? Like it doesn't apply insets if there are no changes to the default ones through the lifecycle.